fe/fe-core/src/main/java/com/starrocks/transaction/TransactionStmtExecutor.java

```diff
@@ -0,0 +1,499 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.transaction;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.primitives.Ints;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.Config;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReportException;
+import com.starrocks.common.LoadException;
+import com.starrocks.common.NoAliveBackendException;
+import com.starrocks.common.StarRocksException;
+import com.starrocks.common.TimeoutException;
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.common.util.concurrent.lock.LockTimeoutException;
+import com.starrocks.lake.LakeTableHelper;
+import com.starrocks.load.EtlJobType;
+import com.starrocks.load.loadv2.InsertLoadJob;
+import com.starrocks.load.loadv2.LoadJob;
+import com.starrocks.load.loadv2.LoadMgr;
+import com.starrocks.metric.MetricRepo;
+import com.starrocks.metric.TableMetricsEntity;
+import com.starrocks.metric.TableMetricsRegistry;
+import com.starrocks.planner.FileScanNode;
+import com.starrocks.planner.OlapScanNode;
+import com.starrocks.planner.ScanNode;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.DefaultCoordinator;
+import com.starrocks.qe.DmlType;
+import com.starrocks.qe.OriginStatement;
+import com.starrocks.qe.QeProcessorImpl;
+import com.starrocks.qe.SessionVariable;
+import com.starrocks.qe.scheduler.Coordinator;
+import com.starrocks.rpc.RpcException;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.service.FrontendOptions;
+import com.starrocks.sql.ast.CreateTableAsSelectStmt;
+import com.starrocks.sql.ast.DeleteStmt;
+import com.starrocks.sql.ast.DmlStmt;
+import com.starrocks.sql.ast.InsertStmt;
+import com.starrocks.sql.ast.StatementBase;
+import com.starrocks.sql.ast.UpdateStmt;
+import com.starrocks.sql.ast.txn.BeginStmt;
+import com.starrocks.sql.ast.txn.CommitStmt;
+import com.starrocks.sql.ast.txn.RollbackStmt;
+import com.starrocks.sql.plan.ExecPlan;
+import com.starrocks.task.LoadEtlTask;
+import com.starrocks.thrift.TExplainLevel;
+import com.starrocks.thrift.TLoadJobType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.starrocks.common.ErrorCode.ERR_TXN_NOT_EXIST;
+
+public class TransactionStmtExecutor {
+    private static final Logger LOG = LogManager.getLogger(TransactionStmtExecutor.class);
+
+    public static void beginStmt(ConnectContext context, BeginStmt stmt) {
+        if (context.getExplicitTxnState() != null) {
+            //Repeated begin does not create a new transaction
```

Comment:
should this be forbidden?

Reply:
mysql not forbidden，we should be compatible with this behavior